src/store/reducer/user.js

```diff
@@ -1,223 +1,266 @@
-        class: '10',
-      };
-
-    case 'C11':
-      const newArray6 = state.subjdata.filter(user => {
-        return (
-          user.class1 == '11' ||
-          user.class2 == '11' ||
-          user.class3 == '11' ||
-          user.class4 == '11'
-        );
-      });
-      return {
-        ...state,
-        users: newArray6,
-        class: '11',
-      };
-
-    case 'C12':
-      const newArray7 = state.subjdata.filter(user => {
-        return (
-          user.class1 == '12' ||
-          user.class2 == '12' ||
-          user.class3 == '12' ||
-          user.class4 == '12'
-        );
-      });
-      return {
-        ...state,
-        users: newArray7,
-        class: '12',
-      };
-
-    case 'LOGIN':
-      return {
-        ...state,
-        loggedin: true,
-        email: action.data.email,
-        currentUser: action.data.name,
-      };
-
-    case 'LOGOUT':
-      return {
-        ...state,
-        loggedin: false,
-        email: '',
-        currentUser: '',
-      };
-
-    default:
-      return state;
-  }
-}
-
-export const getInitalData = () => async dispatch => {
-  try {
-    let all = await axios.get('/users/');
-    // You're dispatching not only the metadata, but also setting isDataInitialized to true, to denote, that data has been loaded
-    dispatch({ type: 'DATA_INITIALIZED', all, isDataInitialized: true });
-  } catch (error) {
-    console.log(error);
-  }
-};
-
-export const getregisteredUserData = data => async dispatch => {
-  try {
-    dispatch({ type: 'LOGIN', data });
-    let userData = await axios.get(`/users/${data.email}`);
-    if (userData.data !== null) {
-      dispatch({ type: 'REGISTERED_USER', userData, registeredUser: true });
-    }
-  } catch (error) {
-    console.log(error);
-  }
-};
-
-const store = createStore(rootReducer, applyMiddleware(thunkMiddleware));
-
-export default store;
+// axios
+//   .get("http://localhost:5000/users/")
+//   .then((response) => {
+//     this.setState({ users: response.data });
+//     console.log(this.state.users);
+//   })
+//   .catch((error) => {
+//     console.log(error);
+//   });
+
+import thunkMiddleware from 'redux-thunk';
+import { createStore, applyMiddleware } from 'redux';
+import axios from 'axios';
+
+const defaultState = {
+  all: {},
+  users: {}, // It should be empty during store init
+  pindata: {},
+  subjdata: {},
+  tutordata: {},
+  isDataInitialized: false, // You can add additional property to denote, that data is not fetched for the first time
+  subject: 'All',
+  class: 'All',
+  currentUser: '',
+  email: '',
+  loggedin: false,
+  registeredUser: false,
+  registeredUserData: {},
+};
+
+function rootReducer(state = defaultState, action) {
+  switch (action.type) {
+    case 'DATA_INITIALIZED':
+      return {
+        ...state,
+        all: action.all.data,
+        users: action.all.data,
+        pindata: action.all.data,
+        subjdata: action.all.data,
+        tutdata: action.all.data,

```

Comment:
It should be tutordata NOT tutdata since you have mentioned tutordata as the default state